anitya/lib/model.py

```diff
@@ -439,6 +439,19 @@ def by_name_and_homepage(cls, session, name, homepage):
         return query.first()
 
     @classmethod
+    def by_version_url(cls, session, version_url):
+        try:
+            query = session.query(
+                cls
+            ).filter(
+                cls.version_url == version_url
+            )
+            return query.one()
```

Comment:
There will be many cases where this will return more than one result, in which case, there is in theory no guarantee about which project you are retrieving